Marlin/Marlin_main.cpp

```diff
@@ -8281,29 +8271,33 @@ inline void gcode_M907() {
 
 #if HAS_CASE_LIGHT
 
-  uint8_t case_light_brightness = 255;
+  uint8_t case_light_brightness;
 
   void update_case_light() {
-    digitalWrite(CASE_LIGHT_PIN, case_light_on != INVERT_CASE_LIGHT ? HIGH : LOW);
-    analogWrite(CASE_LIGHT_PIN, case_light_on != INVERT_CASE_LIGHT ? case_light_brightness : 0);
+    digitalWrite(CASE_LIGHT_PIN, (case_light_brightness >= 128) != INVERT_CASE_LIGHT ? HIGH : LOW);
+    analogWrite(CASE_LIGHT_PIN, INVERT_CASE_LIGHT ? 255 - case_light_brightness : case_light_brightness );
   }
+#endif
 
-#endif // HAS_CASE_LIGHT
 
 /**
- * M355: Turn case lights on/off and set brightness
+ * M355: Turn case lights on/off (and set brightness if a PWM pin is used)
+ *
+ *   S<byte>  Set case light brightness
  *
- *   S<bool>  Turn case light on or off
- *   P<byte>  Set case light brightness (PWM pin required)
  */
 inline void gcode_M355() {
   #if HAS_CASE_LIGHT
-    if (code_seen('P')) case_light_brightness = code_value_byte();
-    if (code_seen('S')) case_light_on = code_value_bool();
-    update_case_light();
-    SERIAL_ECHO_START;
-    SERIAL_ECHOPGM("Case lights ");
-    case_light_on ? SERIAL_ECHOLNPGM("on") : SERIAL_ECHOLNPGM("off");
+    if (code_seen('S')) {
+      case_light_brightness = code_value_byte();
+      update_case_light();
+      if (NOT_ON_TIMER == digitalPinToTimer(CASE_LIGHT_PIN)) {
+        SERIAL_ECHO_START;
+        SERIAL_ECHOPGM("Case lights ");
+        (case_light_brightness >= 128) ? SERIAL_ECHOLNPGM("on") : SERIAL_ECHOLNPGM("off");
```

Comment:
When the pin for the case light is non-PWM then it makes sense that the brightness set by `S` affect `case_light_on`. So…
```cpp
case_light_on = case_light_brightness >= 128;
serialprintPGM(case_light_on ? PSTR("on") : PSTR("off"));
```